### PR TITLE
updating generated frameworks for uniform behaviour

### PR DESCRIPTION
**Pull request by jages**
## Description

We are currently discussing a lot of changes towards code generation. This is one of the first changes towars having a uniform test environment for testing basic framework functionality.

## Motivation and Context

All generated frameworks should have the the same functionality and the same tests on that functionality.

Currently, there a a variety of functions missing from different Frameworks and this PR is addressing those issues.

Feel free to comment with any recommendations towards changes made to the Framework code.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause compatibility issues)
- [ ] DevOps change (fix or alteration in a build pipeline or config file)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: openvalidation-generation/src/main/resources/java/framework.hbs [hunk 1]
```diff
@@ -390,12 +390,16 @@ public List<ValidationRule> Rules = new ArrayList<>();
         return !AT_LEAST_ONE_OF(leftOperand, rightOperand);
     }
 
-    public <T extends Number> Double SUM_OF(T... items)
+    public <T extends Number> BigDecimal SUM_OF(T... items)
```
Comment (sbischoff-ai) [inline openvalidation-generation/src/main/resources/java/framework.hbs]: Would it not be far more elegant to use the interface `Number` as return type and ...

File: openvalidation-generation/src/main/resources/java/framework.hbs [hunk 2]
```diff
@@ -390,12 +390,16 @@ public List<ValidationRule> Rules = new ArrayList<>();
...
         if (items == null || items.length < 1)
-            return 0.0;
+            return new BigDecimal(0);
 
-        return Arrays.stream(items).mapToDouble(i -> i.doubleValue()).sum();
+        BigDecimal value = new BigDecimal(0);
+        for(T item : items){
+            value = BigDecimal.valueOf(item.doubleValue()).add(value);
+        }
+        return value;
```
Comment (sbischoff-ai) [inline openvalidation-generation/src/main/resources/java/framework.hbs]: ... cast `value` to `T` here?

Or even not use BigDecimal at all. Hmm.
Imagine the user used a custom implementation of `Number` in the model, that overloads the add operator?
Maybe I'm overthinking it with the generality here.
This is probably fine.

File: openvalidation-generation/src/main/resources/javascript/framework.hbs [hunk 3]
```diff
@@ -145,4 +145,12 @@ var HUMLFramework = function() {
...
     }
+
+    this.SUM_OF = function(operand){
+        var value = 0;
+        for (let i=0; i<operand.length; i++) {
+            value += operand[i];
+          }
+        return value;
+    }
 }
```
Comment (sbischoff-ai) [inline openvalidation-generation/src/main/resources/javascript/framework.hbs]: Will you implement `this.FIRST` and `this.LAST` in this PR as well?

Comment (jages) [reply openvalidation-generation/src/main/resources/javascript/framework.hbs]: Yes. Maybe it's not very visible, but i will even out the functionality of the three base generated Frameworks. Adding it to the description.

Comment (sbischoff-ai) [reply openvalidation-generation/src/main/resources/javascript/framework.hbs]: I have to confess though, I have just fiddled with the java framework in my Python PR.
I refactored the `getArrayOf` and `where` methods to `GET_ARRAY_OF` and `WHERE`, to be more consistent in the naming of cross-language HUMLFramework functions.
I hope this won't bring up nasty conflicts here.

**Pull request merged**